### PR TITLE
Fix bugs, standarise some docs in UG and DG

### LINKED ISSUE #73: [PE-D][Tester C] Should be able to add a person with the same name, but other different fields
Adding a person with the same name but with a completely different address, email, and number should not be classified as the same person.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: minreiseah/ped#1

### LINKED ISSUE #117: [PE-D][Tester C] Unclear reference to data file in FAQ
While the intent is to refer "empty data file" to `data/addressbook.json`, this is not clearly communicated in the user guide. Less tech-savvy users might not make this connection.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: minreiseah/ped#18

### LINKED ISSUE #119: [PE-D][Tester C] There should be some reasonable limits on the length of inputs
There should be rasonable limits on input lengths, particularly to prevent database overflow, ensure data integrity, and enhance user experience.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: minreiseah/ped#19

### LINKED ISSUE #129: [PE-D][Tester B] Clear can cause a lot of data loss
The user might accidentally call `clear` and remove years worth of information in the app. This is not really probable, but when it happens it'd be catastrophic if the user doesn't have backups.

I think at the very least having a prompt like "Are you sure you want to do this?" would be good.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: miffi/ped#14

### PR DESCRIPTION
**Pull request by guanquann**
- Resolves #117 
- Resolves #119 
- Resolves #129 
- Resolves #73 
- Update use cases in DG
- Rename `showfav` to `listfav` in docs

- Rename data and log file to gourmetgrid. https://github.com/nus-cs2103-AY2324S2/forum/issues/752
- Make default address of contact more sensible
- Update UG styling for standarisation
  - Notes section are now wrapped inside "info" box
  - Reminder section are wrapped inside "tip" box
- Acknowledgement section
- Update class diagram in DG to include `Order`
  - image
  - image
  - image

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -187,7 +185,7 @@ Adds the contacts specified by index as favourites
 Format: `addfav i/INDICES`
 - Adds the contacts at the specified `INDICES` as favourites. The indices refer to comma-separated index numbers (i.e. index, index, index) shown in the displayed person list. Each index **must be a positive integer** 1,2,3, ...
```
Comment (maze508) [inline docs/UserGuide.md:186]: Can you help change this to `contact list`

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@
   - Listing all persons : list
   - Editing a person : edit
```
Comment (maze508) [inline README.md:15]: Can you help change these 2 lines to `contact`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -250,7 +250,7 @@ Below is the sequence diagram for the `addorder` command process:
 
 <puml src="diagrams/AddOrderSequenceDiagram.puml" alt="AddOrderSequenceDiagram" />
 
-### List order Feature
+### List order feature
 
 The `listorder` feature allows users to list all orders associated with a person in the address book, sorted by date in ascending order first, then sorted by order they were added in if date is the same. This is particularly useful for users who wish to track the order history of suppliers efficiently.
```
Comment (maze508) [inline docs/DeveloperGuide.md:255]: Can you help change to `contact` in this sentence

**Review by maze508 — COMMENTED** [review]
some small suggestions, otherwise LGTM

File: docs/UserGuide.md (hunk 1 above)
Comment (guanquann) [reply docs/UserGuide.md:186]: Resolved

**Pull request merged**